hal/common/EventLoop.cpp

```diff
@@ -0,0 +1,62 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef MBED_CONF_RTOS_PRESENT
+
+#include "EventLoop.h"
+#include "Event.h"
+#include "Callback.h"
+
+namespace mbed {
+
+EventLoop::EventLoop(uint32_t size, osPriority priority, uint32_t stack_size, unsigned char *stack_pointer):
+    _queue(size),_thread(priority, stack_size, stack_pointer) {
+}
+
+void EventLoop::start() {
+    _thread.start(callback(this, &EventLoop::dispatch));
+}
+
+void EventLoop::stop() {
+    _thread.terminate();
```

Comment:
@bogdanm Unlike `terminate`, `join` can allows the user to do a proper cleanup. 
Maybe the behavior can be selected with an input parameter:
* `TERMINATE`: terminate right away, current activity is stopped; might lead to resources leak. 
* `STOP_AFTER_CURRENT_EVENT`: Stop the event loop once the current event has been processed. The event loop doesn't accept events after this point. 
* `STOP_WHEN_QUEUE_EMPTY`: The event loop doesn't accept events after this point. Stop when the queue is empty.
* `REQUEST_STOP`: The event loop **accept** events after this point. Stop when the queue is empty. This might never stop